sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala

```diff
@@ -415,6 +417,36 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       }
     }
   }
+
+  private def javaSerializerFor(beanClass: Class[_])(inputObject: Expression): CreateNamedStruct = {
+    JavaTypeInference.serializerFor(inputObject, TypeToken.of(beanClass)) match {
+      case e => CreateNamedStruct(Literal("value") :: e :: Nil)
+    }
+  }
+
+  test("SPARK-23589 ExternalMapToCatalyst should support interpreted execution") {
```

Comment:
Can you also add a test for a `null` key?

Reply:
ok

Reply:
done